src/base/bittorrent/private/resumedatasavingmanager.cpp

```diff
@@ -44,12 +44,16 @@ void ResumeDataSavingManager::save(const QString &filename, const QByteArray &da
     const QString filepath = m_resumeDataDir.absoluteFilePath(filename);
 
     QSaveFile file {filepath};
-    if (file.open(QIODevice::WriteOnly)) {
-        file.write(data);
-        if (!file.commit()) {
-            Logger::instance()->addMessage(QString("Couldn't save data in '%1'. Error: %2")
-                                           .arg(filepath, file.errorString()), Log::WARNING);
-        }
+    if (!file.open(QIODevice::WriteOnly)) {
+        LogMsg(tr("Couldn't save data to '%1'. Error: %2")
+            .arg(filepath, file.errorString()), Log::CRITICAL);
+        return;
+    }
+
+    file.write(data);
+    if (!file.commit()) {
+        LogMsg(tr("Fail to commit data to '%1'. Error: %2")
```

Comment:
I think the message preamble can be the same as the previous one (to avoid any kind of useless implementation details).
Then you can even concatenate conditions since body will be the same.

Reply:
you mean like the following?
```c++
    QSaveFile file {filepath};
    if (!file.open(QIODevice::WriteOnly)
        || (file.write(data) != data.size()) || !file.commit()) {
        LogMsg(tr("Couldn't save data to '%1'. Error: %2")
            .arg(filepath, file.errorString()), Log::CRITICAL);
        return;
    }
```

Reply:
Yes, it is.
Except `return` statement that is redundant.

Reply:
Done.